File: entidades-exemplo/src/main/java/com/fredsonchaves07/domain/services/OrderService.java
package com.fredsonchaves07.domain.services;

import com.fredsonchaves07.domain.entity.Customer;
import com.fredsonchaves07.domain.entity.Order;
import com.fredsonchaves07.domain.entity.OrderItem;

import java.util.List;
import java.util.UUID;

public class OrderService {

    public static int total(List<Order> orders) {
        return orders.stream().reduce(0, (subTotal, order) -> subTotal + order.getTotal(), Integer::sum);
    }

    public static Order placeOrder(Customer customer, List<OrderItem> itens) {
        if (itens.isEmpty()) {
            throw new Error("Oder must have at least one item");
        }
        Order order = new Order(UUID.randomUUID().toString(), customer.getId(), itens);
        customer.addRewardPoints(order.getTotal() / 2);
        return order;
    }
}

File: desafio-fullcycle/src/domain/checkout/service/order.service.ts
import crypto from "crypto";

import { Customer } from "../../customer/entity/customer";
import { Order } from "../entity/order";
import { OrderItem } from "../entity/order_item";

export class OrderService {
  public static calculateTotalPrice(orders: Order[]): number {
    return orders.reduce((total, order) => total + order.total(), 0);
  }

  public static placeOrder(customer: Customer, items: OrderItem[]): Order {
    if (items.length === 0) throw new Error("Items are required");

    const order = new Order(crypto.randomUUID(), customer.id, items);
    customer.addRewardPoints(order.total() / 2);
    return order;
  }
}
